nbcparser/pkg/parser/templates/containerd.toml.gtpl

```diff
@@ -45,7 +45,7 @@ root = "{{.KubeletConfig.GetContainerDataDir}}"{{- end}}
     {{- end}}
     {{- if getIsKrustlet .GetWorkloadRuntime }}
     [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.spin]
-      runtime_type = "io.containerd.spin-v0-3-0.v1"
+      runtime_type = "io.containerd.spin.v2"
```

Comment:
could we add a new agentbakere2e validator to ensure that the containerd config is correct when using krustlet?

https://github.com/Azure/AgentBaker/blob/a9d43fb18e048d580d039de43d00eee22f1ac647/e2e/validation.go#L79

otherwise this lgtm I think

Reply:
Added one